ValheimPlus/GameClasses/Inventory.cs

```diff
@@ -80,4 +80,42 @@ public static void Prefix(string name, ref int w, ref int h)
 			}
 		}
 	}
+
+	/// <summary>
+	/// When merging another inventory, try to merge items with existing stacks.
+	/// </summary>
+	[HarmonyPatch(typeof(Inventory), "MoveAll")]
+	public static class Inventory_MoveAll_Patch
+	{
+		private static void Prefix(ref Inventory __instance, ref Inventory fromInventory)
+		{
+			if (Configuration.Current.Inventory.IsEnabled && Configuration.Current.Inventory.mergeWithExistingStacks)
+			{
+				List<ItemDrop.ItemData> list = new List<ItemDrop.ItemData>(fromInventory.GetAllItems());
+				foreach (ItemDrop.ItemData otherItem in list)
+				{
+					if (otherItem.m_shared.m_maxStackSize > 1)
+					{
+						foreach (ItemDrop.ItemData myItem in __instance.m_inventory)
+						{
+							if (myItem.m_shared.m_name == otherItem.m_shared.m_name && myItem.m_quality == otherItem.m_quality)
+							{
+								int itemsToMove = Math.Min(myItem.m_shared.m_maxStackSize - myItem.m_stack, otherItem.m_stack);
+								myItem.m_stack += itemsToMove;
+								if (otherItem.m_stack == itemsToMove)
+								{
+									fromInventory.RemoveItem(otherItem);
+									break;
+								}
+								else
+								{
+									otherItem.m_stack -= itemsToMove;
```

Comment:
Again, I've not written mods, but does this move all patch also run the default move all after this is run?  I ask as this will likely leave items in the dropped inventory that I'd still want to try to pick up.

Reply:
Ah, I see from your original comment that it sounds like it does your code and then does the default code.

Reply:
https://harmony.pardeike.net/articles/patching-prefix.html#changing-the-result-and-skipping-the-original

A prefix patch with a void return type will always pass on to other patches and then the original method. You can return false to bypass the original method.